nemo_skills/pipeline/train.py

```diff
@@ -31,6 +32,135 @@
+    training_script = "python -m nemo_skills.training.start_sft"
+    extra_arguments = (
+        f" ++model.data.train_ds.file_path='{params.training_data}' "
+        f" ++model.data.validation_ds.file_path='{params.validation_data}' "
+        f" model.restore_from_path={params.nemo_model} " + params.extra_arguments
+    )
+    cmd = (
+        f"export WANDB_API_KEY={os.getenv('WANDB_API_KEY', '')} && "
+        f"export HF_TOKEN={get_token()} && "
+        f"export HYDRA_FULL_ERROR=1 && "
+        f"export PYTHONPATH=$PYTHONPATH:/nemo_run/code && "
+        f"cd /nemo_run/code && "
+        f"echo 'Starting training' && "
+        f"{training_script} "
+        f"    {config_params}"
+        f"    ++model.tensor_model_parallel_size={params.num_gpus} "
+        f"    trainer.devices={params.num_gpus} "
+        f"    trainer.num_nodes={params.num_nodes} "
+        f"    {params.logging_params} "
+        f"    exp_manager.name={params.expname} "
+        f"    exp_manager.explicit_log_dir={params.output_dir}/training "
+        f"    exp_manager.exp_dir={params.output_dir}/training "
+        f"    ++exp_manager.max_time_per_run={params.timeout} "
+        f"    {extra_arguments} "
+    )
+    return cmd
+
+
+def train_dpo(params: TrainingParams) -> str:
+    if params.config_name is None:
+        config_name = "dpo_config"
+    else:
+        config_name = params.config
+    config_params = f"--config-name={config_name} --config-path={params.config_path} "
+    training_script = "python -m nemo_skills.training.start_dpo"
+    extra_arguments = (
+        f" ++model.data.data_prefix.train='[{params.training_data}]' "
+        f" ++model.data.data_prefix.validation='[{params.validation_data}]' "
+        f" ++model.data.data_prefix.test='[{params.validation_data}]' "
+        f" pretrained_checkpoint.restore_from_path={params.nemo_model} " + params.extra_arguments
+    )
+    cmd = (
+        f"export WANDB_API_KEY={os.getenv('WANDB_API_KEY', '')} && "
+        f"export HF_TOKEN={get_token()} && "
+        f"export HYDRA_FULL_ERROR=1 && "
+        f"export PYTHONPATH=$PYTHONPATH:/nemo_run/code && "
+        f"cd /nemo_run/code && "
+        f"echo 'Starting training' && "
+        f"{training_script} "
+        f"    {config_params}"
+        f"    ++model.tensor_model_parallel_size={params.num_gpus} "
+        f"    trainer.devices={params.num_gpus} "
+        f"    trainer.num_nodes={params.num_nodes} "
+        f"    {params.logging_params} "
+        f"    exp_manager.name={params.expname} "
+        f"    exp_manager.explicit_log_dir={params.output_dir}/training "
+        f"    exp_manager.exp_dir={params.output_dir}/training "
+        f"    ++exp_manager.max_time_per_run={params.timeout} "
+        f"    {extra_arguments} "
+    )
+    return cmd
+
+
+def train_rm(params: TrainingParams) -> str:
+    if params.config_name is None:
+        config_name = "dpo_config"
```

Comment:
I guess this is currently unused, but see another comment that we should add the config from nemo aligner here ideally

Reply:
fixed in the most recent refactor of this.